Core/clim-basic/decls.lisp

```diff
@@ -196,6 +196,24 @@
 (defgeneric sheet-occluding-sheets (sheet child))
 (defgeneric map-over-sheets (function sheet))
 
+(defgeneric sheet-name (sheet)
+  (:documentation "McCLIM extension: Return the name of SHEET.
+The returned name is a symbol and does not change.
+For sheets which are also panes, the returned name is identical to the
+pane name.")
+  (:method (sheet) nil))
+(defgeneric sheet-pretty-name (sheet)
+  (:documentation "McCLIM extension: Return the pretty name of SHEET.
+The returned name is a string and may change over time.
+The pretty name usually corresponds to the title of the associated
+window.")
+  (:method (sheet) "(Unnamed sheet)"))
```

Comment:
you may ignore this remark, but I wonder if default method like this wouldn't be better:
```
(if-let ((name (sheet-name sheet)))
  (format nil "~a" name)
  "(Unnamed sheet)")
```